### PR TITLE
[BLOCKED] Avatar builder final improvements

### PR DESCRIPTION
**Pull request by wojtus7**

### CODE REVIEW

File: ios/TransactionListView.swift [hunk 1]
```diff
@@ -50,7 +50,14 @@ class TransactionListView: UIView, UITableViewDelegate {
     }
   }
   @objc func onAvatarPressed(_ sender: UIButton) {
-    self.onAvatarPress([:])
+    tableView.setContentOffset(.zero, animated: true)
+    if tableView.contentOffset.y == CGFloat(0) {
+      self.onAvatarPress([:])
+    } else {
+      DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
```
Comment (brunobar79) [inline ios/TransactionListView.swift]: Can you move the 0.2 to a var at the top specifying what the delay is accounting for?
```
@objc var delayForSomething: TimeInterval = 0.2
```

**Comment by christianbaroni** [comment]
@wojtus7 after I change the avatar background color, if I close the avatar builder and reopen it, the selected color circle is always the light blue

you seeing the same?

**Comment by christianbaroni** [comment]
hm actually it's not always light blue, seems like it could be that it saves your selected color inconsistently

this is in debug mode

**Comment by christianbaroni** [comment]
@wojtus7 going to push some changes. also looked into things more closely, consolidating here to keep track:

1) looks like the react navigation or stack upgrade broke some things
     a) interactions are blocked for 1-2s after closing a stack
     b) keyboard pops up and stays open after closing the swap modal
     c) status bars don't always behave correctly (white when should be black or vice versa)

2) is there a way to use transformOrigin from redash for the emoji stack transition? that was how I had it originally but can't get it working now. did our switch to animated from reanimated make that no longer possible? also, do you know why we switched to animated in the first place?

3) chosen emoji/color are lost after force quitting

4) color circle issues mentioned above (doesn't consistently save after closing avatar builder)

5) often closing the avatar builder feels laggy, looks like it almost just skips the animation (probably low fps), but other times it's smooth. especially noticeable if you open and quickly close. any idea what's causing that?

**Comment by wojtus7** [comment]
@christianbaroni answering question 2:

We switched to animated only in react-navigation API because react-navigation itself switched from reanimated to animated, so to keep any animation we had to switch too. 
I entered the redash library to see how transformOrigin is implemented and it turned out that it's written with reanimated-nodes in mind, so I'm afraid we are not able to use it, till react-navigation won't turn back to reanimated at some point :(

But as far as I dig into it you can make similar results using ```transform: [{matrix: ...}]```

**Comment by wojtus7** [comment]
Also @christianbaroni I tried to fix 5 but it's hard because if anything is going on javascript thread (like updating assets values) it can affect animations written in Animated :(
Plus on very fast taps it is affected by still loading emojis - I can turn on some gesture blocker to make it impossible to go back in the first second to avoid lag because of emoji loading but I'm afraid it won't be any good from UX side since there will be no response for touch ;-;

File: ios/TransactionListView.swift (hunk 1 above)
Comment (wojtus7) [reply ios/TransactionListView.swift]: Resolved 🎉
Sorry, It took me so long I was sure I committed it already, but I actually must have stashed it somewhere along the way :/

**Comment by christianbaroni** [comment]
@wojtus7 any luck with 1a? that's really the only thing blocking merging

**Comment by wojtus7** [comment]
Just created PR to navigation: https://github.com/react-navigation/react-navigation/pull/7871 to resolve 1a 🎉

**Comment by wojtus7** [comment]
Upgraded stack to 2.3.7 but unfortunately still can reproduce 1b sometimes 😭
Will have to dig into that ⛏

**Comment by wojtus7** [comment]
@christianbaroni @jinchung now with new fixed react-navigation combined with an upgraded stack from earlier is should be all good 🥳

**Pull request merged**